pkg/controller/flags.go

```diff
@@ -28,20 +28,22 @@ import (
 	// optimize client
 	fs.StringVar(&velaclient.CachedGVKs, "optimize-cached-gvks", "", "Types of resources to be cached. For example, --optimize-cached-gvks=Deployment.v1.apps,Job.v1.batch . If you have dedicated resources to be managed in your system, you can turn it on to improve performance. NOTE: this optimization only works for single-cluster.")
 	fs.BoolVar(&cache.OptimizeListOp, "optimize-list-op", true, "Optimize ResourceTracker & ApplicationRevision list op by adding index. This will increase the use of memory and accelerate the list operation of ResourceTracker. Default to enable it. If you want to reduce the memory use of KubeVela, you can switch it off.")
+	fs.DurationVarP(&cache.ApplicationRevisionDefinitionCachePruneDuration, "application-revision-definition-storage-prune-duration", "", cache.ApplicationRevisionDefinitionCachePruneDuration, "the duration for running application revision storage pruning")
+	fs.BoolVarP(&cache.OptimizeInformerCache, "optimize-informer-cache", "", cache.OptimizeInformerCache, "if enabled, informer cache will be optimized by removing unnecessary fields will be filtered")
+	fs.BoolVarP(&cache.OptimizeApplicationRevisionDefinitionStorage, "optimize-application-revision-definition-storage", "", cache.OptimizeApplicationRevisionDefinitionStorage, "if enabled, application revision definition cache will be optimized by using reducing duplicated definition storage")
+	fs.BoolVarP(&cache.OptimizeDisableWorkflowContextConfigMapCache, "optimize-disable-workflow-context-configmap-cache", "", cache.OptimizeDisableWorkflowContextConfigMapCache, "if enabled, disable the workflow context's configmap informer cache")
 
 	// optimize functions
 	fs.Float64Var(&resourcekeeper.MarkWithProbability, "optimize-mark-with-prob", 0.1, "Optimize ResourceTracker GC by only run mark with probability. Side effect: outdated ResourceTracker might not be able to be removed immediately. Default to 0.1. If you want to cleanup outdated resource for keepLegacyResource mode immediately, set it to 1.0 to disable this optimization.")
 	fs.BoolVar(&application.DisableAllComponentRevision, "optimize-disable-component-revision", false, "Optimize ComponentRevision by disabling the creation and gc. Side effect: rollout cannot be used. If you don't use rollout trait, you can switch it on to reduce the storage and improve performance.")
 	fs.BoolVar(&application.DisableAllApplicationRevision, "optimize-disable-application-revision", false, "Optimize ApplicationRevision by disabling the creation and gc. Side effect: application cannot rollback. If you don't need to rollback applications, you can switch it on to reduce the storage and improve performance.")
-	fs.BoolVar(&workflow.DisableRecorder, "optimize-disable-workflow-recorder", false, "Optimize workflow recorder by disabling the creation and gc. Side effect: workflow will not record application after finished running. If you do not use VelaUX, you can switch it on to improve performance.")
```

Comment:
does it a break change?

Reply:
No break change. It is an empty flag now.